daemon/logger/jsonfilelog/read.go

```diff
@@ -174,6 +181,17 @@ func followLogs(f *os.File, logWatcher *logger.LogWatcher, notifyRotate chan int
 		fileWatcher.Close()
 	}()
 
+	logWatcherCtx, logWatcherClosed := context.WithCancel(ctx)
+	go func() {
+		select {
+		case <-logWatcher.WatchClose():
+			fileWatcher.Remove(name)
```

Comment:
It's no problem, I had assumed the deferred `fileWatcher.Close()` would have taken care of it but looking at the source it doesn't clear up any open inotify watchers so we would be leaking descriptors. Fix incoming.